File: app/src/main/java/com/example/bmi/Recipe.java
package com.example.bmi;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

public class Recipe extends AppCompatActivity {

    private TextView recipeNameTextView;
    private TextView ingredientsTextView;
    private TextView directionsTextView;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_recipe);


        Intent intent = getIntent();
        String recipeName = intent.getStringExtra("recipe_name");
        String ingredients = intent.getStringExtra("recipe_ingredients");
        String directions = intent.getStringExtra("recipe_directions");


        recipeNameTextView = findViewById(R.id.textView_recipe_name);
        recipeNameTextView.setText(recipeName);

        ingredientsTextView = findViewById(R.id.textView_recipe_ingredients);
        ingredientsTextView.setText(ingredients);

        directionsTextView = findViewById(R.id.textView_recipe_directions);
        directionsTextView.setText(directions);

    }
}

File: app/src/main/java/com/example/bmi/Recipe.kt
package com.example.bmi

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class Recipe : AppCompatActivity() {
    private var recipeNameTextView: TextView? = null
    private var ingredientsTextView: TextView? = null
    private var directionsTextView: TextView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_recipe)
        val intent = intent
        val recipeName = intent.getStringExtra("recipe_name")
        val ingredients = intent.getStringExtra("recipe_ingredients")
        val directions = intent.getStringExtra("recipe_directions")
        recipeNameTextView = findViewById<TextView>(R.id.textView_recipe_name)
        recipeNameTextView?.text = recipeName
        ingredientsTextView = findViewById(R.id.textView_recipe_ingredients)
        ingredientsTextView?.setText(ingredients)
        directionsTextView = findViewById(R.id.textView_recipe_directions)
        directionsTextView?.setText(directions)
    }
}